### PR TITLE
VMware: Add tag facts to vmware_guest_facts

### LINKED ISSUE #46460: Return any Tags Assigned to VM as value in returned dict
##### SUMMARY

Tag assignment not present in response data.

##### ISSUE TYPE
- Feature Idea
Add "tags" key with all assigned tag names as value,
in return json

##### COMPONENT NAME

vmware_vm_facts

##### ADDITIONAL INFORMATION

It would provide capacity to identify if a VM is to be included or excluded from automation,
for any given reason, eg generic configuration for a specific VM may impact the operations of a special case VM, hence the requirement to have capacity to exclude it if possible.

```yaml
- name: Gather all registered virtual machines
  vmware_vm_facts:
    hostname: esxi_or_vcenter_ip_or_hostname
    username: username
    password: password
    uuid: 2341234123412341234123
  delegate_to: localhost
  register: vmfacts
```

##### EXPECTED RESULTS

Tags currently assigned to the VM

##### ACTUAL RESULTS

Tag assignment not present in response data

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #46460

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware_rest_client.py
lib/ansible/modules/cloud/vmware/vmware_guest_facts.py

##### ANSIBLE VERSION

```paste below
2.8-devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @tchernomax @ckotte Could you please review this ? Thanks.

**Review by pdellaert — APPROVED** [review]
LGTM

**Review by tchernomax — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@pdellaert @tchernomax @blambo10 Thanks for the review.

File: lib/ansible/modules/cloud/vmware/vmware_guest_facts.py [hunk 1]
```diff
@@ -120,11 +199,25 @@ def main():
...
+            instance = pyv.gather_facts(vm)
+            if module.params.get('tags'):
+                if not HAS_VCLOUD:
+                    module.fail_json(msg="Unable to find 'vCloud Suite SDK' Python library which is required."
+                                         " Please refer this URL for installation steps"
+                                         " - https://code.vmware.com/web/sdk/60/vcloudsuite-python")
+
+                vm_rest_client = VmwareTag(module)
+                instance.update(
+                    tags=vm_rest_client.get_vm_tags(vm_rest_client.tag_service,
```
Comment (jeking3) [inline lib/ansible/modules/cloud/vmware/vmware_guest_facts.py:211]: The latest pyVmomi 6.7.1 seems to have tag support without going to REST.  I only have a VCSA in my environment (v6.7).  Perhaps we just need to update pyVmomi and don't need the vCloud / REST requirement here.

When I serialize out the contents of a virtual machine I see this now (with just pyVmomi 6.7.1):
```
    "tag": [
        {
            "dynamicProperty": [], 
            "dynamicType": null, 
            "key": "SYSTEM/COM.VMWARE.VIM.VC"
        }
    ], 
```

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_guest_facts.py:211]: OK, does it work for 6.5 and Pyvmomi 6.5 ? I was getting 
```
"tag": []
```

Comment (jeking3) [reply lib/ansible/modules/cloud/vmware/vmware_guest_facts.py:211]: I just tried it with pyVmomi 6.7.1, not sure about any other versions.  Would need to check the pyVmomi project commits to see what's different.  I opened a question for this.  https://github.com/vmware/pyvmomi/issues/733

**Pull request merged**